use-timescale/migration/dual-write-and-backfill.md

```diff
@@ -157,9 +185,101 @@ commit;
 EOF
 ```
 
+<Highlight type="note">
+Background jobs need to be disabled to prevent continuous aggregates from
+updating with incomplete data. Once the migration is complete, we will
+refresh the continuous aggregates for the required range.
+</Highlight>
+
 #### From plain PostgreSQL
 
-TODO
+Dump the database roles from the source database:
+
+```
+pg_dumpall -d "$SOURCE" \
+  --quote-all-identifiers \
+  --roles-only \
+  --file=roles.sql
+```
+
+Timescale services do not support roles with superuser access. If your SQL
+dump includes roles that have such permissions, you'll need to modify the file
+to be compliant with the security model.
+
+You can use the following `sed` command to remove unsupported statements and
+permissions from your roles.sql file:
+
+```
+sed -i -E \
+-e '/CREATE ROLE "postgres";/d' \
+-e '/ALTER ROLE "postgres"/d' \
+-e 's/(NO)*SUPERUSER//g' \
+-e 's/(NO)*REPLICATION//g' \
+-e 's/(NO)*BYPASSRLS//g' \
+-e 's/GRANTED BY "[^"]*"//g' \
+roles.sql
+```
+
+<Highlight type="note">
+This command works only with the GNU implementation of sed (sometimes referred
+to as gsed). For the BSD implementation (the default on macOS), you need to
+add an extra argument to change the `-i` flag to `-i ''`.
+
+To check the sed version, you can use the command sed --version. While the GNU
+version will explicitly identify itself as GNU, the BSD version of sed
+generally doesn't provide a straightforward --version flag and will simply
+output an "illegal option" error.
+</Highlight>
+
+- `CREATE ROLE "postgres"`; and `ALTER ROLE "postgres"`: These statements are
+  removed because they require superuser access, which is not supported
+  by TimescaleDB Cloud.
+
+- `(NO)SUPERUSER` | `(NO)REPLICATION` | `(NO)BYPASSRLS`: These are permissions
+  that require superuser access.
+
+- `GRANTED BY role_specification`: The GRANTED BY clause can also have permissions that
+  require superuser access and should therefore be removed. Note: Per the
+  TimescaleDB documentation, the GRANTOR in the GRANTED BY clause must be the
+  current user, and this clause mainly serves the purpose of SQL compatibility.
+  Therefore, it's safe to remove it.
+
+Dump all plain tables and the TimescaleDB catalog from the source database:
+
+```
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --file=dump.sql
+```
+
+a. `--no-tablespaces` is required because Timescale does not support
+   tablespaces other than the default. This is a limitation.
+b. `--no-owner` is required because tsdbadmin is not a superuser and cannot
+   assign ownership in all cases. This flag means that everything will be
+   owned by the tsdbadmin user in the target regardless of ownership in the
+   source. This is a limitation.
```

Comment:
> This flag means that everything will be owned by the tsdbadmin user in the target

IIF tsdbadmin is used to run pg_restore

Reply:
WDYT about not mentioning tsdbadmin? Maybe something like:

> b. `--no-owner` is required because assigning ownership to certain objects
   could necessitate superuser permissions. Such permissions are not supported
   by Timescale services. When using this flag, the ownership of migrated
   objects will automatically be assigned to the role that is used to run the
   command, regardless of the original ownership in the source database. This is
   a limitation.

Reply:
@alejandrodnm I think that suggestion is good. I've included something like it in my revision.